### ISSUE TITLE
Freemarker checkTemplate default value

### ISSUE THREAD

**Issue by twicksell** [spring-projects/spring-boot#2379]
Any thoughts on setting spring.freemarker.checkTemplateLocation to false by default, or perhaps downgrading it to a warning? An unexpected consequence of upgrading boot was that several of my apps started failing because they didn't have any templates. Unbeknownst to me I was pulling in freemarker from some transitive dependency. Manually setting properties or excluding autoconfigs for something my app doesn't really use seems wrong. 

I think this may apply to velocity as well.

**Comment by wilkinsona** [comment]
The check was originally added as people were trying to use Freemarker, Thymeleaf, whatever but had their templates in the wrong place and were complaining that things weren't working. This check helps them to realise what they've done wrong by failing with an obvious error. Unfortunately, it comes at the cost of tripping up people who are in your situation. There's no obviously right answer here as we can't perfectly satisfy the needs of both camps. Perhaps a warning would be a reasonable compromise, but I suspect people will miss it.

**Comment by snicoll** [comment]
Yep, the warning is pretty much rollbacking what've done IMO. Maybe we could add a global option to disable the check? (i.e. regardless of the template engines you'd have in your classpath)

**Comment by twicksell** [comment]
A global check would make it a bit easier. For example I think the groovy and mustache template check properties were left out of the reference docs. I ended up making a small spring boot module with a PropertySource for my org to reset this default for everyone, so it was actually easier to solve than I thought it was going to be. 

I'm fine with leaving it as it is, but one question does come to mind. What about making it a warning was insufficient to help people identify the issue? Should there be a pronounced "Boot Status Report" immediately after startup letting people know that some features of Boot weren't able to start properly due to things like missing templates? Should there be a /problems actuator? I can totally see how someone might miss a one line warning about templates during the scrolling wall of text at startup, but maybe if we captured and condensed some of that it could be useful. Just thinking out loud. I really want to preserve the aspect of Boot "just working", as opposed to only working after you set some properties you didn't know you needed for features you didn't really want. This is more of a search for opinions than specific feature request, so feel free to close if you guys don't agree.
